egs/wsj/s5/steps/nnet3/components.py

```diff
@@ -24,6 +24,16 @@ def AddInputLayer(config_lines, feat_dim, splice_indexes=[0], ivector_dim=0):
     return {'descriptor': splice_descriptor,
             'dimension': output_dim}
 
+def AddRecurrentInputNodesForLstm(config_lines, name, cell_dim = 0, recurrent_projection_dim = 0):
```

Comment:
Specify that this method is necessary for state preserving training. Briefly describe how the state preservation training works i.e., talk about the reserved node names, talk about the nodes users need to add if they want to write a new RNN with state preservation training.